javalib/src/main/scala/java/io/File.scala

```diff
@@ -264,7 +264,7 @@ class File(_path: String) extends Serializable with Comparable[File] {
 
   def list(filter: FilenameFilter): Array[String] =
     if (!isDirectory() || !canRead()) {
-      null
+      Array.empty[String]
```

Comment:
I agree, I will change code which calls this list and doesn't check for null (some internal implementation)